graphql/subcourse/mutations.ts

```diff
@@ -80,27 +80,39 @@ export class MutateSubcourseResolver {
     ): Promise<GraphQLModel.Subcourse> {
         const course = await getCourse(courseId);
         await hasAccess(context, 'Course', course);
-
-        const { joinAfterStart, minGrade, maxGrade, maxParticipants, allowChatContactParticipants, allowChatContactProspects, groupChatType } = subcourse;
-        const result = await prisma.subcourse.create({
-            data: {
-                courseId,
-                published: false,
-                joinAfterStart,
-                minGrade,
-                maxGrade,
-                maxParticipants,
-                allowChatContactParticipants,
-                allowChatContactProspects,
-                groupChatType,
+        const courseInstructorAssociation = await prisma.course_instructors_student.findFirst({
+            where: {
+                courseId: courseId,
+                studentId: studentId,
             },
         });
+        const isCourseSharedOrOwned = !!courseInstructorAssociation || course.shared;
+        if (!isCourseSharedOrOwned) {
+            logger.error(`Subcourse(${courseId}) is not shared or Student(${studentId}) is not an instructor of this subcourse`);
+            throw new PrerequisiteError('This subcourse is not shared or you are not the instructor of this subcourse!');
+        }
 
-        const student = await getSessionStudent(context, studentId);
-        await prisma.subcourse_instructors_student.create({ data: { subcourseId: result.id, studentId: student.id } });
+        if (isCourseSharedOrOwned) {
```

Comment:
unnecessary if

Reply:
Thanks, you are right since I already throw an error in line 92 when the if condition isnt met :)